tests/30rooms/12thirdpartyinvite.pl

```diff
@@ -4,377 +4,199 @@
 use Protocol::Matrix qw( encode_base64_unpadded sign_json );
 use SyTest::Identity::Server;
 
-my @user_preparers = local_user_preparers( 2 );
-
 my $crypto_sign = Crypt::NaCl::Sodium->sign;
 
 my $DIR = dirname( __FILE__ );
 
+my $invitee_email = 'lemurs@monkeyworld.org';
+
 test "Can invite existing 3pid",
-   requires => [ @user_preparers ],
+   requires => [ local_user_preparers( 2 ) ],
 
    do => sub {
       my ( $inviter, $invitee ) = @_;
 
-      my $invitee_email = 'marmosets@monkeyworld.org';
       my $invitee_mxid = $invitee->user_id;
 
       my $room_id;
 
-      my $stub_id_server = SyTest::Identity::Server->new;
-      $stub_id_server->{bindings}{$invitee_email} = $invitee_mxid;
-      $loop->add( $stub_id_server );
-      $stub_id_server->listen(
-         host    => "localhost",
-         service => "",
-         extensions => [qw( SSL )],
-         # Synapse currently only talks IPv4
-         family => "inet",
-
-         SSL_cert_file => "$DIR/../../keys/tls-selfsigned.crt",
-         SSL_key_file => "$DIR/../../keys/tls-selfsigned.key",
-      )->then( sub {
-         my ( $listener ) = @_;
-
-         my $sock = $listener->read_handle;
-         my $id_server = sprintf "%s:%d", $sock->sockhostname, $sock->sockport;
-
-         matrix_create_and_join_room( [ $inviter ], visibility => "private" )
-         ->then( sub {
-            ( $room_id ) = @_;
-            do_request_json_for( $inviter,
-               method => "POST",
-               uri    => "/api/v1/rooms/$room_id/invite",
-
-               content => {
-                  id_server    => $id_server,
-                  medium       => "email",
-                  address      => $invitee_email,
-                  display_name => "Cute things",
-               },
-            );
-         })->then( sub {
-            matrix_get_room_state( $inviter, $room_id,
-               type      => "m.room.member",
-               state_key => $invitee_mxid,
-            )->on_done( sub {
-               my ( $body ) = @_;
-
-               log_if_fail "Body", $body;
-               $body->{membership} eq "invite" or
-                  die "Expected invited user membership to be 'invite'";
+      start_id_server()->then(sub {
+         my ( $id_server ) = @_;
+
+         $id_server->bind_identity( undef, "email", $invitee_email, $invitee )->then(sub {
```

Comment:
Eh; actually now I'm using a wide enough github browser window, perhaps it's not so bad. Maybe leave it ;)